### ISSUE TITLE
Zone Masters are set wrong via REST API

### ISSUE THREAD

**Issue by klaus3000** [PowerDNS/pdns#6451]
From the docs:

> PowerDNS supports multiple masters. For ... SQL based backends, list all master servers separated by commas in the 'master' field of the domains table.

But ws-auth adds the masters with space separated, and one extra space at the end. I am not familiar with the REST API, hence no pull request. But this is how I fixed it for me, feel free to use the patch if it is correct.

```
diff --git a/pdns/ws-auth.cc b/pdns/ws-auth.cc
index 74a9f7e..9f131fb 100644
--- a/pdns/ws-auth.cc
+++ b/pdns/ws-auth.cc
@@ -504,16 +504,17 @@ static void gatherComments(const Json container, const DNSName& qname, const QTy
 }

 static void updateDomainSettingsFromDocument(const DomainInfo& di, const DNSName& zonename, const Json document) {
-  string zonemaster;
+  vector<string> zonemaster;
   for(auto value : document["masters"].array_items()) {
     string master = value.string_value();
     if (master.empty())
       throw ApiException("Master can not be an empty string");
-    zonemaster += master + " ";
+    zonemaster.push_back(master);
   }

   di.backend->setKind(zonename, DomainInfo::stringToKind(stringFromJson(document, "kind")));
-  di.backend->setMaster(zonename, zonemaster);
+  di.backend->setMaster(zonename, boost::join(zonemaster, ","));

   if (document["soa_edit_api"].is_string()) {
     di.backend->setDomainMetadataOne(zonename, "SOA-EDIT-API", document["soa_edit_api"].string_value());
```

Thanks
Klaus

**Comment by Habbie** [comment]
space and tab are also supported in SQL, to be clear, but I agree we should be consistent between docs and code etc.

**Comment by klaus3000** [comment]
Any chances to have this fixed in 4.2?
